packages/lodestar/src/network/reqresp/reqResp.ts

```diff
@@ -55,73 +67,45 @@ export class ReqResp implements IReqResp {
 
   start(): void {
     this.controller = new AbortController();
-    for (const method of Object.values(Method)) {
-      for (const encoding of Object.values(ReqRespEncoding)) {
-        this.libp2p.handle(
-          createRpcProtocol(method, encoding),
-          async ({connection, stream}: {connection: Connection; stream: ILibP2pStream}) => {
-            const peerId = connection.remotePeer;
-
-            // TODO: Do we really need this now that there is only one encoding?
-            // Remember the prefered encoding of this peer
-            if (method === Method.Status) {
-              this.peerMetadata.encoding.set(peerId, encoding);
-            }
-
-            try {
-              await handleRequest(
-                {config: this.config, logger: this.logger, libp2p: this.libp2p},
-                this.onRequest.bind(this),
-                stream,
-                peerId,
-                method,
-                encoding,
-                this.controller.signal,
-                this.respCount++
-              );
-              // TODO: Do success peer scoring here
-            } catch {
-              // TODO: Do error peer scoring here
-              // Must not throw since this is an event handler
-            }
-          }
-        );
-      }
+    for (const [method, version, encoding] of protocolsSupported) {
+      this.libp2p.handle(
+        formatProtocolId(method, version, encoding),
+        this.getRequestHandler({method, version, encoding})
```

Comment:
Move logic to a separate function `getRequestHandler()` to keep code more modular